src/plugin/oauth/index.js

```diff
@@ -2166,32 +2167,48 @@ class BaseOauthPlugin extends BasePlugin {
       ) {
         await plugin.set_introspection_cache(session_id, response);
       }
+      plugin.server.logger.verbose("Assertions: PASSED introspect_access_token")
+    } else {
+      plugin.server.logger.verbose("Assertions: SKIPPED introspect_access_token")
     }
 
-    if (
-      pluginStrategy == PLUGIN_STRATEGY_OIDC &&
-      plugin.config.assertions.id_token
-    ) {
-      let idToken;
-      idToken = jwt.decode(tokenSet.id_token);
-      let idTokenValid = await plugin.id_token_assertions(idToken);
+
+    if (plugin.config.assertions.id_token) {
+      if (!tokenSet.id_token) {
+        plugin.server.logger.verbose("Missing id_token")
+        return false
+      }
+
+      const idToken = jwt.decode(tokenSet.id_token);
+      const idTokenValid = await plugin.id_token_assertions(idToken);
       if (!idTokenValid) {
+        plugin.server.logger.verbose("Invalid id_token")
         return false;
       }
+      plugin.server.logger.verbose("Assertions: PASSED id_token")
+    } else {
+      plugin.server.logger.verbose("Assertions: SKIPPED id_token")
     }
 
-    if (
-      pluginStrategy == PLUGIN_STRATEGY_OIDC &&
-      plugin.config.assertions.access_token
-    ) {
-      let accessToken;
-      accessToken = jwt.decode(tokenSet.access_token);
```

Comment:
Was this the line that was blowing up previously? Or was it something else?

Reply:
Yes, I believe the decode function doesn't handle undefined inputs